app/controllers/bikes_controller.rb

```diff
@@ -153,51 +153,107 @@ def create
 
   def edit
     @page_errors = @bike.errors
-    @edit_template = edit_templates[params[:page]].present? ? params[:page] : edit_templates.keys.first
+    @edit_templates = edit_templates
+
+    is_valid, @edit_template = target_edit_template(params[:page], edit_templates.keys)
+
+    if !is_valid
+      redirect_to edit_bike_url(@bike, page: @edit_template) and return
+    end
+
     if @edit_template == "photos"
-      @private_images = PublicImage.unscoped.where(imageable_type: "Bike").where(imageable_id: @bike.id).where(is_private: true)
+      @private_images =
+        PublicImage
+          .unscoped
+          .where(imageable_type: "Bike")
+          .where(imageable_id: @bike.id)
+          .where(is_private: true)
     end
+
     render "edit_#{@edit_template}".to_sym
   end
 
   def update
     begin
-      @bike = BikeUpdator.new(user: current_user, bike: @bike, b_params: permitted_bike_params.as_json, current_ownership: @current_ownership).update_available_attributes
+      @bike = BikeUpdator.new(
+        user: current_user,
+        bike: @bike,
+        b_params: permitted_bike_params.as_json,
+        current_ownership: @current_ownership,
+      ).update_available_attributes
     rescue => e
       flash[:error] = e.message
     end
+
     @bike = @bike.decorate
+
     if @bike.errors.any? || flash[:error].present?
       edit and return
     else
       flash[:success] = "Bike successfully updated!"
       return if return_to_if_present
-      redirect_to edit_bike_url(@bike, page: params[:edit_template]) and return
-    end
-  end
 
-  def edit_templates_hash
-    stolen_type = @bike.recovered ? "Recovery" : "Theft"
-    hash = {
-      root: "Bike Details",
-      photos: "Photos",
-      drivetrain: "Wheels + Drivetrain",
-      accessories: "Accessories + Components",
-      ownership: "Transfer Ownership",
-      groups: "Groups + Organizations",
-      remove: "Hide or Delete",
-      stolen: (@bike.stolen ? "#{stolen_type} details" : "Report Stolen or Missing"),
-    }
-    # To make stolen the first key if bike is stolen. using as_json for string keys instead of sym
-    (@bike.stolen ? hash.to_a.rotate(-1).to_h : hash).as_json
+      # When reporting a bike as stolen, redirect to the recovery form.
+      # The recovery form redirects to the stolen form and is handled by
+      # CoffeeScript in pages/bikes/edit_stolen.coffee
+      target_template = (params[:edit_template] == "report_stolen") ?
+        "report_recovered" : params[:edit_template]
+      redirect_to edit_bike_url(@bike, page: target_template) and return
+    end
```

Comment:
Currently the form simply reloads and is rerendered with the recovery form replacing the stolen form and vice versa. This reproduces that behavior by redirecting with the appropriate query param, but it may be more desirable from a UX standpoint to redirect to the base form.

Reply:
I think each template should be unique and uniquely named. So there is a "report stolen" form and then a "theft details" form.

Reply:
I'm not sure if I answered your question correctly? Let me know if I missed the point.

Reply:
> I think each template should be unique and uniquely named. 

That's the setup on this branch. The behavior I'm maintaining from master is that when you submit the "report stolen" form, you then see the "report recovery" form.

Reply:
>  the "report stolen" form simply reloads and is rerendered with the recovery form replacing the stolen form and vice versa. 

I expressed that poorly but hopefully the GIF disambiguates -- I meant to say that when the user marks a bike as stolen, the next thing they see is the "mark as recovered" form.

Reply:
Ahhh. Yes, I understand now. 

I think the behavior you suggested, of redirecting to to the base form, would be an improvement. Your choice of when/whether to make that update.

Reply:
Cool, that simplifies the implementation a bit too -- updated in f7c5fb4